### ISSUE TITLE
Filtering related set with escapable characters

### ISSUE THREAD

**Issue by dhbahr** [django-salesforce/django-salesforce#260]
As per https://docs.djangoproject.com/en/2.2/topics/db/queries/#escaping-percent-signs-and-underscores-in-like-statements django auto-escapes especial characters like `%` or `_` on LIKE query sets (`startswith`, `endswith`, and the likes of those).

I am experiencing in 1.0 a weird behavior, namely, on a custom object I try to get a specific attachment, using the expected attachment name, like so:

```
from salesforce import models
...
class CustomObject(models.Model):
  ...
  attachment_name_auto = models.CharField(...)
  ...
  def get_attachment(self):
    return self.attachment_set.filter(name__startswith=self.attachment_name_auto)
```

The problem lies with the naming of the attachment, it follows a pattern like `<formatted_date_value>_<contact_name>_<formatted_time_range>`, and it causes the queryset to return empty.

I have confirmed that using any part of the name between underscores yields results, only when an underscore is included, the behavior becomes buggy.

Is there a specific way I should escape the string beforehand or is this a bug?

**Comment by dhbahr** [comment]
Hah .. I just realized that I had not upgraded the staging environment to 1.0 and was still in 0.9. it does work properly in 1.0, so please disregard this entirely.

**Comment by hynekcer** [comment]
It is also a duplicate of #254 where you find details, why it was so and why the current better solution can't be perfect.

Simply: Salesforce uses a simplified escaping and requires it. That doesn't allow to search an exact '%' or '_' in `LIKE` expressions. It is better to search a string with a wildcard instead of to find nothing and even to be not able to construct any less restrictive string. That is the substance of the fix in v1.0.

The result can be filtered exactly by Python if necessary finally.
